### PR TITLE
Add support for piping to STDIN by checking if input device is terminal

### PR DESCRIPTION
**Pull request by JesterOrNot**
This should allow for stuff like

```bash
echo ls | gosh
```

### CODE REVIEW

**Comment by JesterOrNot** [comment]
@RDIL what do you think?

File: main.go [hunk 1]
```diff
@@ -46,5 +44,22 @@ func main() {
...
+			shell.Shell(quiet)
+		} else {
+			reader := bufio.NewReader(os.Stdin)
+			text, _ := reader.ReadString('\n')
+			shell.Evaluate(strings.Trim(text, "\n"))
+		}
 	}
 }
+
+func find(source []string, value string) bool {
```
Comment (RDIL) [inline main.go:58]: I think I made a function for this - maybe we should join them together in a utils file

Comment (JesterOrNot) [reply main.go:58]: Good plan!

Comment (RDIL) [reply main.go]: I know I used it in another file so you will need to track that down also

Comment (JesterOrNot) [reply main.go]: Done!

**Comment by JesterOrNot** [comment]
Take another look?

**Comment by JesterOrNot** [comment]
Going to merge, followup PRs can be made as needed.

**Pull request merged**